// @multi-publisher/shared/src/fieldDefinitions.ts
import type { FieldDefinition } from "./fields";

export const fieldDefinitions: FieldDefinition[] = [
  {
    key: "title",
    label: "Título",
    type: "text",
    required: true,
    section: "root",
    group: "general",
    placeholder: "Ej. iPhone 13 128GB en buen estado",
  },
  {
    key: "description",
    label: "Descripción",
    type: "textarea",
    required: true,
    section: "root",
    group: "general",
    placeholder: "Describe el producto, estado, accesorios, defectos, etc.",
  },
  {
    key: "price",
    label: "Precio",
    type: "number",
    required: true,
    section: "root",
    group: "general",
    placeholder: "Ej. 450",
  },
  {
    key: "category",
    label: "Categoría",
    type: "select",
    required: true,
    section: "root",
    group: "general",
    options: [
      { label: "Electrónica", value: "electronics" },
      { label: "Moda", value: "fashion" },
      { label: "Hogar", value: "home" },
      { label: "Motor", value: "motor" },
    ],
  },
  {
    key: "subcategory",
    label: "Subcategoría",
    type: "text",
    section: "root",
    group: "general",
    placeholder: "Ej. smartphones",
  },
  {
    key: "condition",
    label: "Estado",
    type: "select",
    required: true,
    section: "root",
    group: "details",
    options: [
      { label: "Nuevo", value: "new" },
      { label: "Como nuevo", value: "like_new" },
      { label: "Buen estado", value: "good" },
      { label: "Aceptable", value: "fair" },
      { label: "Usado", value: "used" },
    ],
  },
  {
    key: "location",
    label: "Ubicación",
    type: "text",
    required: true,
    section: "root",
    group: "details",
    placeholder: "Ej. Madrid",
  },
  {
    key: "images",
    label: "Imágenes",
    type: "image-picker",
    required: true,
    section: "root",
    group: "images",
    helpText: "Busca por nombre o examina la carpeta configurada.",
  },

  {
    key: "brand",
    label: "Marca",
    type: "text",
    section: "attributes",
    group: "details",
    categories: ["electronics", "fashion", "motor"],
  },
  {
    key: "model",
    label: "Modelo",
    type: "text",
    section: "attributes",
    group: "details",
    categories: ["electronics", "motor"],
  },
  {
    key: "color",
    label: "Color",
    type: "text",
    section: "attributes",
    group: "details",
  },
  {
    key: "storage",
    label: "Almacenamiento",
    type: "text",
    section: "attributes",
    group: "details",
    categories: ["electronics"],
    placeholder: "Ej. 128GB",
  },

  {
    key: "shippingEnabled",
    label: "Permitir envío",
    type: "checkbox",
    section: "portalSettings",
    portal: "wallapop",
    group: "shipping",
    portals: ["wallapop"],
  },
  {
    key: "adType",
    label: "Tipo de anuncio",
    type: "select",
    section: "portalSettings",
    portal: "milanuncios",
    group: "portal",
    portals: ["milanuncios"],
    options: [
      { label: "Venta", value: "sale" },
      { label: "Compra", value: "wanted" },
    ],
  },
];
